quarkchain/cluster/shard.py

```diff
@@ -128,6 +128,44 @@ def broadcast_tx_list(self, tx_list):
             self.shard_state.root_tip, self.shard_state.header_tip, header_list
         )
 
+    async def handle_get_minor_block_header_list_with_skip_request(self, request):
+        if request.branch != self.shard_state.branch:
+            self.close_with_error("Wrong branch from peer")
+        if request.limit <= 0 or request.limit > 2 * MINOR_BLOCK_HEADER_LIST_LIMIT:
+            self.close_with_error("Bad limit")
+        if request.type != 0 and request.type != 1:
+            self.close_with_error("Bad type value")
+
+        if request.type == 1:
+            block_height = request.get_height()
+        else:
+            block_hash = request.get_hash()
+            block_header = self.shard_state.db.get_minor_block_header_by_hash(block_hash)
+            if block_header is None:
+                return GetMinorBlockHeaderListResponse(self.shard_state.root_tip, self.shard_state.header_tip, [])
+
+            # Check if it is canonical chain
+            block_height = block_header.height
+            if self.shard_state.db.get_minor_block_header_by_height(block_height) != block_header:
```

Comment:
note that our minor chain may have higher probability of being on a fork

Reply:
True.  But it depends on the locality of the shard BPs.   This is more a server-side code for future optimization, and we may not replace the current sync code at the moment.